File: AndroidXPatternsApp/jmvvm-app/src/main/java/com/woowrale/jmvvm/di/components/MainComponent.java
package com.woowrale.jmvvm.di.components;

import com.woowrale.jmvvm.di.modules.MainModule;
import com.woowrale.jmvvm.di.modules.ServicesModule;
import com.woowrale.jmvvm.ui.base.BaseActivity;
import com.woowrale.jmvvm.ui.search.home.SearchActivity;
import com.woowrale.jmvvm.ui.search.local.LocalSearchActivity;
import com.woowrale.jmvvm.ui.search.remote.RemoteSearchActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {MainModule.class, ServicesModule.class})
public interface MainComponent {

    void inject(BaseActivity activity);

    void inject(SearchActivity activity);

    void inject(LocalSearchActivity activity);

    void inject(RemoteSearchActivity activity);

}

File: AndroidXPatternsApp/kmvp-app/src/main/java/com/woowrale/kmvp/di/components/MainComponent.kt
package com.woowrale.kmvp.di.components

import com.woowrale.kmvp.di.modules.MainModule
import com.woowrale.kmvp.di.modules.ServicesModule
import com.woowrale.kmvp.ui.base.BaseActivity
import com.woowrale.kmvp.ui.search.home.SearchActivity
import com.woowrale.kmvp.ui.search.local.LocalSearchActivity
import com.woowrale.kmvp.ui.search.remote.RemoteSearchActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [MainModule::class, ServicesModule::class])
interface MainComponent {

    fun inject(activity: BaseActivity)

    fun inject(activity: SearchActivity)

    fun inject(activity: LocalSearchActivity)

    fun inject(activity: RemoteSearchActivity)

}